### PR TITLE
Added a Puppeteer Can Shapeshift Setting.

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: Modules/GuessManager.cs [hunk 1]
```diff
@@ -189,12 +189,6 @@ public static bool GuesserMsg(PlayerControl pc, string msg, bool isUI = false)
                     else pc.ShowPopUp(GetString("GuessDoctor"));
                     return true;
                 }
-                if (Medic.ProtectList.Contains(target.PlayerId) && !Medic.GuesserIgnoreShield.GetBool())
-                {
-                    if (!isUI) Utils.SendMessage(GetString("GuessShielded"), pc.PlayerId);
-                    else pc.ShowPopUp(GetString("GuessShielded"));
-                    return true;
```
Comment (Tommy-XL) [inline Modules/GuessManager.cs:196]: Your code removes the setting for Medic

File: Modules/GuessManager.cs [hunk 2]
```diff
@@ -504,9 +498,6 @@ public static bool GuesserMsg(PlayerControl pc, string msg, bool isUI = false)
 
                     Utils.NotifyRoles(isForMeeting: false, NoCache: true);
 
-                    if (dp.Is(CustomRoles.Medic))
-                        Medic.IsDead(dp);
-
```
Comment (Tommy-XL) [inline Modules/GuessManager.cs:509]: The same thing, but when Medicr is guessed

File: Modules/CustomRolesHelper.cs [hunk 3]
```diff
@@ -181,8 +181,8 @@ internal static class CustomRolesHelper
                 CustomRoles.Tracefinder => CustomRoles.ScientistTOHE,
                 CustomRoles.SpeedBooster => CustomRoles.CrewmateTOHE,
                 CustomRoles.Dictator => CustomRoles.CrewmateTOHE,
-                CustomRoles.Snitch => CustomRoles.CrewmateTOHE,
-                CustomRoles.Marshall => CustomRoles.CrewmateTOHE,
+                CustomRoles.Snitch => CustomRoles.EngineerTOHE,
+                CustomRoles.Marshall => CustomRoles.EngineerTOHE,
```
Comment (Tommy-XL) [inline Modules/CustomRolesHelper.cs:185]: Do all these roles really need to be done under the base of the Engineer?

File: Modules/OptionHolder.cs [hunk 4]
```diff
@@ -966,6 +967,8 @@ public static void Load()
             .SetValueFormat(OptionFormat.Seconds);
         SetupRoleOptions(3800, TabGroup.ImpostorRoles, CustomRoles.Miner);
         SetupRoleOptions(3900, TabGroup.ImpostorRoles, CustomRoles.Puppeteer);
+        PuppeteerCanShapeshift = BooleanOptionItem.Create(3712, "PuppeteerCanShapeshift", true, TabGroup.ImpostorRoles, false)
+            .SetParent(CustomRoleSpawnChances[CustomRoles.Puppeteer]);
```
Comment (Tommy-XL) [inline Modules/OptionHolder.cs:971]: "PuppeteerCanShapeshift" Should also be added to string.csv

**Comment by Loonie-Toons** [comment]
What's the point in this setting?

**Comment by Tommy-XL** [comment]
However, your code adds customization and removes the required code from other roles at the same time

**Comment by Loonie-Toons** [comment]
I can very easily do this myself without the issues, its just too powerful.

**Pull request closed without merging**